File: lib/src/models/printer_info.dart
/// PrinterInfo model class
/// This class is used to represent the information of a printer.
/// The information of the printer can be used to determine the printer
/// model, port name, mac address, emulation, and port settings.
class PrinterInfo {
  PrinterInfo({
    required this.portName,
    required this.modelName,
    required this.macAddress,
    required this.emulation,
    required this.portSettings,
  });

  factory PrinterInfo.fromJson(Map<String, dynamic> json) => PrinterInfo(
        portName: json["portName"],
        modelName: json["modelName"],
        macAddress: json["macAddress"],
        emulation: json["emulation"],
        portSettings: json["portSettings"],
      );

  final String portName;
  final String modelName;
  final String macAddress;
  final String emulation;
  final String portSettings;
}

File: android/src/main/kotlin/com/iconica/flutter_thermal_printer/models/PrinterInfo.kt
package com.iconica.flutter_thermal_printer.models

import com.starmicronics.starioextension.StarIoExt

/**
 * This class represents the printer information.
 *
 * @property portName The port name of the printer.
 * @property modelName The model name of the printer.
 * @property macAddress The MAC address of the printer.
 * @property emulation The emulation of the printer.
 * @property portSettings The port settings of the printer.
 *
 * @author Mark Kiepe
 * @since 1.0.0
 * @version 1.0.0
 */
class PrinterInfo(
    val portName: String,
    val modelName: String,
    val macAddress: String,
    val emulation: StarIoExt.Emulation,
    val portSettings: String,
) {

    /**
     * This method converts the printer information to a map.
     *
     * @return Map<String, Any> The printer information as a map.
     */
    fun toMap(): Map<String, Any> {
        return mapOf(
            "portName" to portName,
            "modelName" to modelName,
            "macAddress" to macAddress,
            "emulation" to emulation.toString(),
            "portSettings" to portSettings
        )
    }
}
